services/src/main/java/com/chatbot/services/ChatServiceControllers/HangoutsAuth.java

```diff
@@ -0,0 +1,48 @@
+package com.chatbot.services.ChatServiceControllers;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.Map;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.googleapis.auth.oauth2.GooglePublicKeysManager;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+
+import org.springframework.stereotype.Component;
+
+import com.chatbot.services.ChatServiceConstants;
+
+// Authenticate requests at the chat service endpoint
+
+@Component
+public class HangoutsAuth {
+
+  // verify the request using header params
+  public void verifyRequest(Map<String, String> headers)
+    throws IOException, GeneralSecurityException {
+    final String AUDIENCE = System.getenv("projectNumber");
```

Comment:
lowercase.
Same for other

Reply:
updated

Reply:
Arey..I meant "AUDIENCE" should be lowercase as it is not static final. The other thing was fine